extmod/vfs_posix_file.c

```diff
@@ -92,18 +91,21 @@ mp_obj_t mp_vfs_posix_file_open(const mp_obj_type_t *type, mp_obj_t file_in, mp_
         }
     }
 
-    o->base.type = type;
-
     mp_obj_t fid = file_in;
 
     if (mp_obj_is_small_int(fid)) {
+        // When opening an existing file description, don't close it on deletion
+        mp_obj_vfs_posix_file_t *o = m_new_obj(mp_obj_vfs_posix_file_t);
```

Comment:
description -> descriptor

And please use `mp_obj_malloc()`.

Reply:
Thanks, fix pushed as suggested!